exporter/kafkaexporter/factory_test.go

```diff
@@ -122,15 +122,81 @@ func TestWithMarshalers(t *testing.T) {
 	})
 }
 
-type customMarshaler struct {
+type customTracesMarshaler struct {}
+
+var _ TracesMarshaler = (*customTracesMarshaler)(nil)
+
+func (c customTracesMarshaler) Marshal(_ pdata.Traces, topic string) ([]*sarama.ProducerMessage, error) {
+	panic("implement me")
```

Comment:
Not exactly sure why this should panic in test code, what are you trying to achieve here?